planning/behavior_path_planner/include/behavior_path_planner/utils/safety_checker/safety_checker.hpp

```diff
@@ -0,0 +1,274 @@
+using geometry_msgs::msg::Point;
+using geometry_msgs::msg::Pose;
+using tier4_autoware_utils::LineString2d;
+using tier4_autoware_utils::Polygon2d;
+
+struct TargetObjectIndices
+{
+  std::vector<size_t> current_lane{};
+  std::vector<size_t> target_lane{};
+  std::vector<size_t> other_lane{};
+};
+
+struct ObjectTypesToCheck
+{
+  bool check_car{true};
+  bool check_truck{true};
+  bool check_bus{true};
+  bool check_trailer{true};
+  bool check_unknown{true};
+  bool check_bicycle{true};
+  bool check_motorcycle{true};
+  bool check_pedestrian{true};
+};
+
+struct PredictedPolygons
+{
+  std::vector<double> check_durations;
+  std::vector<Pose> poses;
+  std::vector<Polygon2d> polygons;
+};
+
+// struct PredictedPolygon
+// {
+//   double check_time;
+//   Pose pose;
+//   Polygon2d polygon;
+// };
+
+struct RSSparams
+{
+  double rear_vehicle_reaction_time{0.0};
+  double rear_vehicle_safety_time_margin{0.0};
+  double lateral_distance_max_threshold{0.0};
+  double longitudinal_distance_min_threshold{0.0};
+  double longitudinal_velocity_delta_time{0.0};
+};
+
+struct SafetyCheckParams
+{
+  // Trajectory generation parameters
+  double backward_lane_length{200.0};
+  double prediction_time_resolution{0.5};
+  double target_velocity{4.0};
+  double forward_path_length{300.0};
+
+  // Prediction options
+  bool use_predicted_path_outside_lanelet{false};
+  bool use_all_predicted_path{false};
+
+  // Object types to check for collisions
+  ObjectTypesToCheck object_types_to_check;
+
+  // Buffer and thresholds
+  double lateral_buffer{0.2};
+  double ego_nearest_dist_threshold{3.0};
+  double ego_nearest_yaw_threshold{1.046};
+
+  // Time window for collision checking
+  double check_start_time{4.0};
+  double check_end_time{10.0};
+
+  std::shared_ptr<RouteHandler> route_handler{std::make_shared<RouteHandler>()};
+
+  // Stopping and acceleration parameters
+  double stopping_time{1.0};
+  double acc_till_target_velocity{1.0};
+  double expected_front_deceleration{-1.0};
+  double expected_rear_deceleration{-1.0};
+
+  RSSparams rss_params{};
+
+  // Vehicle dimensions
+  vehicle_info_util::VehicleInfo vehicle_info{};
+  double vehicle_width{1.83};
+
+  // Debug marker publishing option
+  bool publish_debug_marker{false};
+};
+
+struct SafetyCheckData
+{
+  // Route and dynamic objects information
+  PredictedObjects::ConstSharedPtr dynamic_objects;
+  TargetObjectIndices & dynamic_objects_indices;
+  lanelet::ConstLanelets current_lanes{};
+  lanelet::ConstLanelets target_lanes{};
+  // lanelet::ConstLanelets reference_lanes{};
+
+  marker_utils::CollisionCheckDebug collision_check_debug;
+};
+
+namespace safety_checker
+{
+
+class SafetyChecker
+{
+public:
+  explicit SafetyChecker(
+    const std::shared_ptr<SafetyCheckParams> & safety_check_params,
+    const std::shared_ptr<SafetyCheckData> & safety_check_data)
+  : safety_check_params_{safety_check_params}, safety_check_data_{safety_check_data}
+  {
+  }
```

Comment:
add a constructor with only params to make an instance in the module constructor.

Reply:
and updateSafetyCheckData function